### ISSUE TITLE
Stationary structures should not be affected by Acceleration or Inhibitor Zones

### ISSUE THREAD

**Issue by Joshua-Leibold** [Ahli/sc2xml#114]
With the exception of flying Terran structures, there is no reason for structures to be affected by Acceleration/Inhibitor Zones. The particle effects generated upon gaining the Acceleration/Inhibitor Behaviours cause unnecessary clutter and performance drain, the clutter especially because the particles scale with the size of the objects and structures are quite large. It is particularly bad if the zones are placed near mining bases. It is also slightly confusing - I've seen more than one player ask whether or not the structures are affected in some other way as indicated by the effect, such as accelerated unit training speed or energy regeneration

Adding a remove validator to the effects should suffice as a solution

BEFORE: 
https://i.gyazo.com/4f57b99b1319466ce54db95f4ff4563c.mp4

AFTER:
https://i.gyazo.com/20cd781e4ae28ff4cb6e06b324f89af8.mp4

**Comment by Joshua-Leibold** [comment]
**XML Changes:**

```
<CValidatorCombine id="IsFlyingTerranBuilding">
        <ResultFailed value="CantTargetThatUnit"/>
        <CombineArray value="NotFlyingStarport"/>
        <CombineArray value="NotFlyingBarracks"/>
        <CombineArray value="NotFlyingCommandCenter"/>
        <CombineArray value="NotFlyingFactory"/>
        <CombineArray value="NotFlyingOrbitalCommand"/>
        <Negate value="1"/>
    </CValidatorCombine>

<CValidatorCombine id="IsUnitOrFlyingTerranStructure">
        <CombineArray value="NotStructure"/>
        <CombineArray value="IsFlyingTerranBuilding"/>
    </CValidatorCombine>

<CEffectApplyBehavior id="InhibitorZoneTemporalField">
        <ValidatorArray value="IsUnitOrFlyingTerranStructure"/>
    </CEffectApplyBehavior>

<CEffectApplyBehavior id="InhibitorZoneFlyingTemporalField">
        <ValidatorArray value="IsUnitOrFlyingTerranStructure"/>
    </CEffectApplyBehavior>

<CEffectApplyBehavior id="AccelerationZoneTemporalField">
        <ValidatorArray value="IsUnitOrFlyingTerranStructure"/>
    </CEffectApplyBehavior>

<CEffectApplyBehavior id="AccelerationZoneFlyingTemporalField">
        <ValidatorArray value="IsUnitOrFlyingTerranStructure"/>
    </CEffectApplyBehavior>
```

**Comment by Joshua-Leibold** [comment]
Woops, accidentally closed the issue when posting those XML changes

**Comment by Ahli** [comment]
I would simplify the validator a bit (shorter and does not need to compare multiple unit IDs; only flying aspect):

``` xml
    <CValidatorCombine id="noStructureOrFlyingStructure">
        <CombineArray value="NotStructure"/>
        <CombineArray value="TargetIsNotGround"/>
    </CValidatorCombine>

    <CEffectApplyBehavior id="InhibitorZoneTemporalField">
        <ValidatorArray value="noStructureOrFlyingStructure"/>
    </CEffectApplyBehavior>
    <CEffectApplyBehavior id="InhibitorZoneFlyingTemporalField">
        <ValidatorArray value="noStructureOrFlyingStructure"/>
    </CEffectApplyBehavior>
    <CEffectApplyBehavior id="AccelerationZoneTemporalField">
        <ValidatorArray value="noStructureOrFlyingStructure"/>
    </CEffectApplyBehavior>
    <CEffectApplyBehavior id="AccelerationZoneFlyingTemporalField">
        <ValidatorArray value="noStructureOrFlyingStructure"/>
    </CEffectApplyBehavior>
```

**Comment by Joshua-Leibold** [comment]
Successfully patched live with 5.0.13
